src/blocks/file-blocks/edit/index.tsx

```diff
@@ -0,0 +1,192 @@
+          isLoading ? "opacity-50 cursor-default" : ""
+        }`}
+        onSubmit={async (e) => {
+          e.preventDefault();
+          setIsLoading(true);
+          const res = await axios.post("/api/openai-edit", {
+            instruction: instruction,
+            input: content,
+          });
+          setNewContent(res.data);
+          setIsLoading(false);
+        }}
+      >
+        <label className="font-normal">
+          How would you like to edit this code?
+        </label>
+        <div className="flex items-center mt-1">
+          <input
+            className="w-full p-2 form-control"
+            type="text"
+            value={instruction}
+            disabled={isLoading}
+            onChange={(e) => {
+              setInstruction(e.target.value);
+            }}
+          />
+          <button disabled={isLoading} className="btn ml-1 self-stretch">
+            {newContent ? "Re-generate modified code" : "Get modified code"}
+            <span className="ml-2">
+              <RocketIcon />
+            </span>
+          </button>
+        </div>
+      </form>
+
+      <div className="flex items-end px-5 py-2">
+        {newContent && (
+          <div className="w-full flex justify-between">
+            <div className="text-gray-500">Proposed code</div>
+            <button
+              className="btn btn-primary"
+              onClick={() => {
+                onRequestUpdateContent(newContent);
+              }}
+            >
+              Save changes
+            </button>
+          </div>
+        )}
+      </div>
+
+      {newContent ? (
+        <div className="col-span-2">
+          <div className="w-full">
+            {hunks?.[0]?.hunks?.map((hunk: Hunk) => (
+              <HunkComponent
+                key={hunk.content}
+                hunk={hunk}
+                language={language}
+              />
+            ))}
+          </div>
+        </div>
+      ) : (
+        <pre className="px-5 py-3 text-left">
+          <SyntaxHighlighter
+            language={syntaxHighlighterLanguageMap[language] || "javascript"}
+            useInlineStyles={false}
+            showLineNumbers={false}
+            lineNumberStyle={{ opacity: 0.45 }}
+            className="!bg-transparent"
+            wrapLines
+            wrapLongLines
+          >
+            {content}
+          </SyntaxHighlighter>
+        </pre>
+      )}
+    </div>
+  );
+}
+
+const syntaxHighlighterLanguageMap = {
+  JavaScript: "javascript",
+  TypeScript: "typescript",
+} as Record<string, string>;
+
+const HunkComponent = ({ hunk, language }: { hunk: any; language: string }) => {
+  console.log(hunk);
+  return (
+    <pre className="px-5 py-3 text-left">
+      {hunk.changes.map((change: Hunk["change"], i: number) => (
+        <Change key={i} change={change} language={language} />
+      ))}
+    </pre>
+  );
+};
+
+const Change = ({ change, language }: { change: Hunk; language: string }) => {
```

Comment:
this is cool, I think the diff view in `blocks` doesn't do syntax highlighting

Reply:
you're right! it's the same library, which is honestly a pain to work with. I took it a step farther here, but ended up doing the styling & rendering myself and am probably missing some edge cases (seeing how large their codebase is).